testsuite/tests/kuadrant/authorino/operator/tls/conftest.py

```diff
@@ -0,0 +1,119 @@
+"""Conftest for all TLS-enabled tests"""
+from urllib.parse import urlparse
+
+import pytest
+
+from testsuite.certificates import CFSSLClient, Certificate
+from testsuite.openshift.client import OpenShiftClient
+from testsuite.openshift.httpbin import TLSEnvoyHttpbin
+
+
+@pytest.fixture(scope="session")
+def create_secret(blame, request):
+    """Creates TLS secret from Certificate"""
+    def _create_secret(openshift: OpenShiftClient, certificate: Certificate, name):
+        secret_name = blame(name)
+        request.addfinalizer(lambda: openshift.delete_selector(envoy_cert))
+        envoy_cert = openshift.create_tls_secret(secret_name, certificate)
+        return secret_name
+    return _create_secret
+
+
+@pytest.fixture(scope="session")
+def cfssl(testconfig):
+    """CFSSL client library"""
+    client = CFSSLClient(binary=testconfig["cfssl"])
+    if not client.exists:
+        pytest.skip("Skipping CFSSL tests as CFSSL binary path is not properly configured")
+    return client
+
+
+@pytest.fixture(scope="session")
+def wildcard_domain(openshift):
+    """
+    Hostname of the upstream certificate sent to be validated by APIcast
+    May be overwritten to configure different test cases
+    """
+    hostname = urlparse(openshift.api_url).hostname
+    return "*.apps." + hostname.split(".", 1)[1]
+
+
+@pytest.fixture(scope="session")
+def authorino_domain(openshift):
+    """
+    Hostname of the upstream certificate sent to be validated by APIcast
+    May be overwritten to configure different test cases
+    """
+    return f"*.{openshift.project}.svc.cluster.local"
+
+
+@pytest.fixture(scope="session")
+def envoy_authority(wildcard_domain, cfssl):
+    """Authority for all certificates that envoy should accept"""
+    return cfssl.create_ca("backend-ca", hosts=[wildcard_domain])
+
+
+@pytest.fixture(scope="session")
+def invalid_authority(wildcard_domain, cfssl):
```

Comment:
It's not invalid per say probably better term would be not accepted, if you come up with better differentiation than valid/invalid it would be great :D